client/cli/src/commands/run_cmd.rs

```diff
@@ -80,16 +81,22 @@ pub struct RunCmd {
 	/// Listen to all RPC interfaces.
 	///
 	/// Same as `--rpc-external`.
-	#[structopt(long = "unsafe-rpc-external")]
+	#[structopt(long)]
 	pub unsafe_rpc_external: bool,
 
-	/// Don't deny potentially unsafe RPCs when listening on external interfaces.
+	/// RPC methods to expose.
 	///
-	/// Default is false. This allows exposing RPC methods publicly (same as `--unsafe-{rpc,ws}-external` )
-	/// but will allow doing so even on validator nodes, which is prohibited by default.
-	/// Please do this if you know what you're doing.
-	#[structopt(long = "unsafe-rpc-expose")]
-	pub unsafe_rpc_expose: bool,
+	/// Defaults to `Safe` (allow only a safe subset of RPC methods) if RPC is
```

Comment:
You describe the possible values, but not the `Default` variant.

Reply:
I find it strikes a good balance without being very explicit - variants of `Safe` and `Unsafe` are explained (which can be explicitly picked and the behaviour is as explained) and `Default`, well, defaults to one of them, depending on the situation as listed.

Do you have any concrete example on how you'd like to improve it?

Reply:
The docs don't specify what happens with `Default` though, right?

Reply:
You say that it defaults to `Safe`, which is not true. It will default to `Safe`, when the RPC port is exposed externally. 

The user can see all the values this CLI accepts by looking into `--help` and the user also sees the default value. The docs should describe that this argument is doing, so that the user can decide which value he wants.

```
Declare which RPC methods should be exposed under different conditions:

- `Default`: Expose safe & unsafe RPC methods when the RPC is only reachable on localhost. If the RPC port is reachable on all interfaces only the safe RPC methods are exposed.
- `Safe`: Only expose safe RPC methods under any conditions.
- `Unsafe`: Expose safe & unsafe RPC methods under any condition.
```

Something like this?

Reply:
```rust
/// RPC methods to expose.
///
/// The `Unsafe` variant allows every RPC method to be called, while the
/// `Safe` variant only allows for its safe subset.
/// If `Default` variant is specified, it acts as `Safe` if RPC is served
/// externally, e.g. when `--{rpc,ws}-external` is passed, or as `Unsafe` otherwise.
```

Does this read better?

Reply:
@Xanewok sounds good to me.

Reply:
@bkchr I liked your list approach, so I went with that instead (Also needed `verbatim_doc_comment` or the list would get wrapped, otherwise).

This now renders as:
```
        --rpc-methods <METHOD SET>
            RPC methods to expose.
            
            - `Unsafe`: Exposes every RPC method.
            - `Safe`: Exposes only a safe subset of RPC methods, denying unsafe RPC methods.
            - `Default`: Acts as `Safe` if RPC is served externally, e.g. when `--{rpc,ws}-external` is passed,
              otherwise acts as `Unsafe`. [default: Default]  [possible values: Default, Safe, Unsafe]
```

Now that I read about it, the Default variant name does seem off... Especially if structopt injects the `[default: Default`] for us.

Maybe something like `Protected` would be a better name? Or should we just leave it for now?

Reply:
`Auto` sounds good as it handles it for you automatically :D

Reply:
(`auto` is also more common for a CLI)